examples/clusterrole-per-ns/test.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+cleanup() {
+  set +e
+  echo "Clean up..."
+  kubectl delete -f my-namespace.yaml
+  kubectl delete -f clusterrole-per-ns.yaml
+  kubectl delete configmap clusterrole-per-ns-controller -n metacontroller
+}
+trap cleanup EXIT
+
+set -ex
+
+echo "Install controller..."
+kubectl create configmap clusterrole-per-ns-controller -n metacontroller --from-file=sync.py
+kubectl apply -f clusterrole-per-ns.yaml
+
+echo "Create a Namespace..."
+kubectl apply -f my-namespace.yaml
+
+echo "Wait for per-ns Service..."
+until [[ "$(kubectl get clusterrole my-namespace -o 'jsonpath={.metadata.name}')" == "my-namespace" ]]; do sleep 1; done
```

Comment:
It would be good to verify that garbage collection works after the namespace is deleted.

For the service-per-ns case, that may be tricky to verify because there is a separate namespace controller that deletes everything within the namespace being deleted. Can you think of any relatable example of a cluster-scoped decorator creating namespaced attachments, that attaches them to something other than a Namespace object?

Reply:
I can't think of any "real" world use-cases but can try to come up with something contrived.. I started out using nodes but reverted to namespaces since they seemed more logical but I can try something that creates a service for every node in the Default namespace.